### PR TITLE
Cleanup duplicate CDR logic from `pm_icecon`

### PR DESCRIPTION
**Pull request by trey-stafford**
Title. 

Also updates the dev documentation, removes other OBE elements, and does some minor reorg.

Confirmed that the CLI works to generate sample idecdr  data file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,82 +46,7 @@ TODO
...
-```
-
-#### Type-checking
-
-Use `mypy` to run static typechecking
-
-```
-$ mypy
-```
+See [doc/development.md](doc/development.md) for more information.
```
Comment (sc0tts) [inline README.md:49]: Pulling this information into a sub-document is  a *great* idea!

File: doc/development.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## Running tests / generating sample data
+
+Can run unit and integration tests -- or manual subsets of -- with:
+    
+    ./scripts/run_ecdr_pytest.sh
+
+Create an Initial Daily ECDR file with e.g.,:
+
+    ./scripts/cli.sh idecdr --date 2021-04-05 --hemisphere north --resolution 12 --output-dir /tmp/
```
Comment (sc0tts) [inline doc/development.md:83]: Somehow, I want to have a "todo" that eventually, I would like the specification at the command line to be:  date, grid (grid includes projection, resolution, and bounds), and TBtype (TB type includes source and methodology for getting those TBs onto the grid)

File: seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py [hunk 3]
```diff
@@ -61,97 +51,22 @@ def sample_idecdr_dataset_sh():
...
-
-    pmicecon_conc_ds = pmi_amsr2_cdr(
-        date=test_date,
-        hemisphere=test_hemisphere,
-        resolution=test_resolution,
-    )
-    return pmicecon_conc_ds
-
-
-@pytest.fixture(scope="session")
```
Comment (sc0tts) [inline seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py:81]: Oh man, I'm sad to see this pytest.fixture with "session" scope trick disappear.  I was so happy to find out how to implement that type of logic....

File: doc/development.md (hunk 2 above)
Comment (trey-stafford) [reply doc/development.md:83]: 4bedc4f

File: seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py (hunk 3 above)
Comment (trey-stafford) [reply seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py:81]: Still some tests in this module with this decorator :)

**Pull request merged**